### ISSUE TITLE
Saturation of Hyper-Parameters for real world datasets

### ISSUE THREAD

**Issue by ktiwari9** [SheffieldML/GPy#505]
I have a dataset [ozone_data.txt](https://github.com/SheffieldML/GPy/files/1016813/ozone_data.txt) where the 1st 2 columns are latitude and longitude information and the rest of the columns are measurements stored per time step for the corresponding location. Thus, the input X to the GP must be 3D [Lat,Long,Time] and the output Y is 1D [Measurement].

Input X: 59 X 3
Target Y: 59X1

Minimum working Example for GP Inference:
`k = GPy.kern.rbf(input_dim=3, variance=1., lengthscale=1., ARD=True)`
`m = GPy.models.GPRegression(X,Y,k)`
`m.optimize(m.optimize(messages=False,max_f_eval = 50))`

And the hyperparameters are saturated throughout the learning phase -

The list of hyper-parameters for each training iteration are - [signal variance, lengthscale 1, lengthscale 2, lengthscale 3, noisevariance]
 
[[  1.49999999e+01   1.47775673e+00   1.27478835e+00   1.00000000e+00    1.00000000e-02]
 [  1.49999986e+01   4.99998705e+00   1.00000536e+00   1.00000000e+00    1.00000000e-02]
 [  1.49999986e+01   4.99998714e+00   1.00000020e+00   1.00000000e+00    1.00000000e-02]
 [  1.49999985e+01   4.99998942e+00   1.00000235e+00   1.00000000e+00    1.00000000e-02]
 [  1.49999983e+01   4.99999125e+00   1.00001391e+00   1.00000000e+00    1.00000000e-02]
 [  1.50000000e+01   4.99999993e+00   4.57851635e+00   1.00000000e+00    1.00000000e-02]] 

Why are the hyper-parameters never changing much ? Essentially the values that I am seeing are the constraints that I have set for my Hyper-parameters. Is there something I have am missing here?

**Comment by mzwiessele** [comment]
It seems the model wants to go outside the range of your set constraints. Some options: You could try to normalize inputs and outputs, initialize the lengthscale and Gaussian_vairance small, to ensure a data fit in the beginning. You can plot the kernel matrix and see if you have good interactions between your samples. You want a covariance matrix encompassing all possible values throughout the matrix and not a diagonal or uniform matrix.
